dozer-cli/src/simple/build/contract/mod.rs

```diff
@@ -239,4 +240,17 @@ where
     serde_json::from_reader(file).map_err(BuildError::FailedToLoadExistingContract)
 }
 
+impl PartialEq<Contract> for Contract {
+    fn eq(&self, other: &Contract) -> bool {
+        self.version == other.version
```

Comment:
This worries me a little that when we add new fields to `Contract` we'll forget to update `PartialEq` implementation.

Maybe we can do
```rust
let Contract {
  version: self_version,
  pipeline: self_pipeline,
  endpoint: self_endpoints,
} = self;
let Contract {
  version: other_version,
  pipeline: other_pipeline,
  endpoint: other_endpoints,
} = other;
```

And then compare them.

Reply:
Or maybe wrap pipeline in a newtype and impl partialeq on that and derive it on Contract?

Reply:
Even better

Reply:
The new type is called `PipelineContract` by the way. It's already a alias type.